extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -128,9 +130,40 @@ void coreServiceFilter(BuildProducer<CamelServiceFilterBuildItem> filterBuildIte
         }
 
         @BuildStep
-        void serviceInfoTransformers(BuildProducer<CamelServiceInfoTransformerBuildItem> mapperBuildItems) {
-            mapperBuildItems.produce(
-                    new CamelServiceInfoTransformerBuildItem(CamelServiceInfoTransformers::configurer));
+        void coreServices(
+                ApplicationArchivesBuildItem archives,
+                BuildProducer<CamelServiceBuildItem> services) {
+
+            /* Registered only */
+            CamelSupport.propertyFiles(
+                    archives,
+                    "META-INF/services/org/apache/camel/component",
+                    "META-INF/services/org/apache/camel/language",
+                    "META-INF/services/org/apache/camel/dataformat")
+                    .filter(entry -> entry.getValue().getProperty("class") != null)
+                    .map(entry -> CamelServiceBuildItem.registeredOnly(entry.getKey(), entry.getValue().getProperty("class")))
+                    .forEach(services::produce);
+
+            /* Configurers need some name transformation */
+            CamelSupport.propertyFiles(
+                    archives,
+                    "META-INF/services/org/apache/camel/configurer")
+                    .filter(entry -> entry.getValue().getProperty("class") != null)
+                    .map(entry -> CamelServiceBuildItem.registeredOnly(
```

Comment:
As future evolution, may be better not to expose such instances to the user through the registry, maybe better to try to bind them to the components at build time

Reply:
I wonder how can we cover the case you mentioned on the chat - manually bind a second instance of a component under a different name?

Reply:
I mean that we should not even bind them to the registry but scan the registry for components and bind related configurers through a recorder during STATIC_INIT

Reply:
> the registry for components and bind related configurers through a recorder during STATIC_INIT

Yes, that would be easy to do. But when a user adds a second instance of a component under a different name at runtime, our configurer setting code would not catch it.

Reply:
yep but as the second instance is probably configured at runtime, the component will fallback to the "factory finder" way which is reasonable